library-udf/src/main/java/org/apache/iotdb/library/util/LinearRegression.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.library.util;
+
+import java.util.Arrays;
+
+/**
+ * This class offers function to do linear regression. Please use org.apache.commons.math3 for
+ * regular usage.
+ */
+public class LinearRegression {
+  double[] x, y, e, yhead;
+  int n;
+  double sumx, sumy, xbar, ybar, xxbar, yybar, xybar;
+  double beta1, beta0, rss, ssr, R2, svar, svar1, svar0;
+
+  public LinearRegression(double[] a, double[] b) {
+    x = a.clone();
```

Comment:
Does it need to check n should be larger than 0?

Reply:
Checks are added in data profiling part. There is no usage in series discovery functions.